readme: more VS Code Extension info. (#7916)
iterative/dvc: README.rst

#. **Track** experiments in your local Git repo (no servers needed).

> #. **Compare** any data, code, parameters, modelhttps://www.youtube.com/watch?v=IYhVmD-_wRometrics, or performance plots

Reviewer: What happened here 😄 ?

Author: Consistent formatting (in the places I edited). 1 sentence = 1 line

Author: Lol I just realized the hero rat video got in. Gone now

How DVC works
=============

    We encourage you to read our `Get Started
    <https://dvc.org/doc/get-started>`_ guides to better understand what DVC

>     does and how it can fit your scenarios.

Author: Not directly related but we should probably move this to AFTER the **Quick start** sections?

Author: - [ ] BTW it's missing info about Experiment Management (only covers Data Mgmt and Pipelines rn)

Reviewer: Yes on both counts! It's on my personal to-do list along with a host of other readme improvements.

Author: https://github.com/iterative/dvc/pull/7917

>     We encourage you to read our `Get Started
>     <https://dvc.org/doc/get-started>`_ guides to better understand what DVC
>     does and how it can fit your scenarios.

Author: ```suggestion
We encourage you to read our `Get Started <https://dvc.org/doc/get-started>`_ guide to better understand what DVC
is and how it can fit your scenarios.
```

The DVC pipelines (computational graph) feature connects code and data together. It is possible to explicitly
specify all steps required to produce a model: input dependencies including data, commands to run,
and output information to be saved. See the quick start sections below or
the `Get Started <https://dvc.org/doc/get-started>`_ tutorial to learn more.

> Quick start (Visual Studio Code)

Reviewer 2: I think it's a bit extreme. There should be a single Quick Start, it should be about CLI. We can put it somewhere as a note, or as a separate section in the README (not a second Quick Start though).

Author: K

The DVC pipelines (computational graph) feature connects code and data together. It is possible to explicitly
specify all steps required to produce a model: input dependencies including data, commands to run,
and output information to be saved. See the quick start sections below or
the `Get Started <https://dvc.org/doc/get-started>`_ tutorial to learn more.

Quick start (Visual Studio Code)
================================

#. `Install DVC <https://dvc.org/doc/install>`_ on your system.
#. Install the `DVC extension <https://marketplace.visualstudio.com/items?itemName=Iterative.dvc>`_ for VS Code.
#. Follow the **Get Started** page that pops up!

|VS Code Extension Overview|

Quick start (command line)
==========================

>     Please read our `Command Reference <https://dvc.org/doc/command-reference>`_ for a complete list.
> 
> A common CLI workflow includes:

Author: ```suggestion
Please read `Get Started <https://dvc.org/doc/get-started>`_ guide for a full version. Common workflow commands include:
```

+-----------------------------------+----------------------------------------------------------------------------+

> | Task                              | Terminal                                                                   |

Author: ```suggestion
| Step                              | Command                                                                    |
```

+===================================+============================================================================+
| Track data                        | | ``$ git add train.py``                                                   |
|                                   | | ``$ dvc add images.zip``                                                 |
+-----------------------------------+----------------------------------------------------------------------------+

> | Connect code and data             | | ``$ dvc run -n prepare -d images.zip -o images/ unzip -q images.zip``    |

Author: ```suggestion
| Connect code and data by commands | | ``$ dvc run -n prepare -d images.zip -o images/ unzip -q images.zip``    |
```

and output information to be saved. See the quick start section below or
the `Get Started <https://dvc.org/doc/get-started>`_ tutorial to learn more.

Quick start
===========
Quick start (CLI)
=================

    Please read our `Command Reference https://dvc.org/doc/command-reference>`_
    for a full reference.

> Please read `Get Started <https://dvc.org/doc/get-started>`_ guide for a full version. Common workflow commands include:

Reviewer: Why get rid of this? I know it's above, but considering we will want to swap places, it would be good to link to get started early.

Author: WIP ⏳

Author: This was removed because there's 2 Quick starts now, this one is specific to command line so I linked to the Command Reference docs instead.

Installation
============

There are several ways to install DVC: in VS Code; using ``snap``, ``choco``, ``brew``, ``conda``, ``pip``; or with an OS-specific package.
Full instructions are `available here <https://dvc.org/doc/get-started/install>`_.

> Visual Studio Code Extension

Reviewer 2: maybe put the gif here?

Author: Done.

   snap install dvc --classic

This corresponds to the latest tagged release.

> Add ``--beta`` for the latest tagged release candidate, or ``--edge`` for the latest ``main`` version.

Author: ```suggestion
Add ``--beta`` for the latest tagged release candidate,
or ``--edge`` for the latest ``main`` version.
```

> Chocolatey (Windows)
> --------------------

Author: ```suggestion
Choco (Chocolatey/Windows)
--------------------------
```

|Choco|

.. code-block:: bash

   choco install dvc

> Brew (mac OS)
> -------------

Author: ```suggestion
Brew (Homebrew/Mac OS)
----------------------
```

|Brew|

.. code-block:: bash

   brew install dvc

> Anaconda (Any platform)
> -----------------------

Author: ```suggestion
Conda (Anaconda)
----------------
```

   conda install -c conda-forge mamba # installs much faster than conda
   mamba install -c conda-forge dvc

Depending on the remote storage type you plan to use to keep and share your data, you might need to install optional dependencies: `dvc-s3`, `dvc-azure`, `dvc-gdrive`, `dvc-gs`, `dvc-oss`, `dvc-ssh`.

PyPI (Python)
-------------

|PyPI|

.. code-block:: bash

   pip install dvc

Depending on the remote storage type you plan to use to keep and share your data, you might need to specify one of the optional dependencies: ``s3``, ``gs``, ``azure``, ``oss``, ``ssh``. Or ``all`` to include them all.

> The command should look like this: ``pip install dvc[s3]`` (in this case AWS S3 dependencies such as ``boto3`` will be installed automatically).

Reviewer: ```suggestion
The command should look like this: ``pip install 'dvc[s3]'`` (in this case AWS S3 dependencies such as ``boto3`` will be installed automatically).
```

Might be better to include quotes since failing to do so can cause issues on some shells (zsh).

   conda install -c conda-forge mamba # installs much faster than conda
   mamba install -c conda-forge dvc

> Depending on the remote storage type you plan to use to keep and share your data, you might need to install optional dependencies: `dvc-s3`, `dvc-azure`, `dvc-gdrive`, `dvc-gs`, `dvc-oss`, `dvc-ssh`.

Author: ```suggestion
Depending on the remote storage type you plan to use to keep and share your data, you might need to
install optional dependencies: `dvc-s3`, `dvc-azure`, `dvc-gdrive`, `dvc-gs`, `dvc-oss`, `dvc-ssh`.
```

> PyPI (Python)
> -------------

Author: ```suggestion
pip (PyPI)
----------
```

|PyPI|

.. code-block:: bash

   pip install dvc

> Depending on the remote storage type you plan to use to keep and share your data, you might need to specify one of the optional dependencies: ``s3``, ``gs``, ``azure``, ``oss``, ``ssh``. Or ``all`` to include them all.
> The command should look like this: ``pip install 'dvc[s3]'`` (in this case AWS S3 dependencies such as ``boto3`` will be installed automatically).

Author: ```suggestion
Depending on the remote storage type you plan to use to keep and share your data, you might need to specify
one of the optional dependencies: ``s3``, ``gs``, ``azure``, ``oss``, ``ssh``. Or ``all`` to include them all.
The command should look like this: ``pip install dvc[s3]`` (in this case AWS S3 dependencies such as ``boto3``
will be installed automatically).
```

To install the development version, run:

.. code-block:: bash

   pip install git+git://github.com/iterative/dvc

> Package (Platform-specific)
> ---------------------------

Author: ```suggestion
Package
-------
```

|Packages|

> Self-contained packages for Linux, Windows, and Mac are available.
> The latest version of the packages can be found on the GitHub `releases page <https://github.com/iterative/dvc/releases>`_.

Author: ```suggestion
Self-contained packages for Linux, Windows, and Mac are available. The latest version of the packages
can be found on the GitHub `releases page <https://github.com/iterative/dvc/releases>`_.
```
